src/themes/dojo/comboBox.m.css

```diff
@@ -1,55 +1,115 @@
+	position: relative;
+	width: 100%;
 }
 
 .root * {
 	box-sizing: border-box;
 }
 
-.controls {
-	position: relative;
+.root input {
+	line-height: var(--line-height-base);
+	min-width: 200px;
+	padding: var(--spacing-regular) calc(var(--spacing-regular) * 6) var(--spacing-regular) var(--spacing-regular);
+	width: 100%;
 }
 
-.input {
-	width: 100%;
-	padding-right: 2em;
+.trigger {
+	appearance: none;
+	background-color: transparent;
+	border-radius: 0;
+	border: var(--border-width) solid transparent;
+	border-left: var(--border-width) solid var(--color-border);
+	bottom: var(--border-width);
+	color: var(--color-text-faded);
+	cursor: pointer;
+	display: block;
+	font-size: 0;
+	line-height: var(--line-height-base);
+	outline: 0;
+	padding: var(--spacing-regular) calc(var(--spacing-regular) + 4);
+	position: absolute;
+	right: 0;
+	top: var(--border-width);
+	transition: border-color var(--transition-duration) var(--transition-easing);
+	z-index: 1;
 }
 
-.input::-ms-clear {
-	display: none;
+.trigger i {
+	font-size: var(--font-size-base);
+	line-height: var(--line-height-base);
 }
 
-.clearable {
-	padding-right: 4em;
+.trigger:focus {
+	border-color: var(--color-highlight);
 }
 
-.results {
+.dropdown {
+	background-color: var(--color-background);
+	border: var(--border-width) solid var(--color-border);
+	box-shadow: var(--box-shadow-dimensions-small) var(--color-box-shadow);
+	opacity: 0;
 	position: absolute;
+	transform-origin: 0 0;
+	transform: rotateX(90deg);
+	transition: all var(--transition-duration) var(--transition-easing);
 	width: 100%;
-	max-height: var(--select-menu-height);
-	overflow-y: auto;
-	background: var(--dojo-white);
-	border: var(--border-thin) solid var(--dojo-shade-gray);
-	z-index: 1;
+	z-index: 10;
 }
 
-.result {
+.open .dropdown {
+	opacity: 1;
+	transform: rotateX(0);
+}
+
+.option {
+	background-color: var(--color-background);
+	border: var(--border-width) solid transparent;
+	cursor: pointer;
+	line-height: var(--line-height-base);
+	padding: var(--spacing-regular);
+	transition: background var(--transition-duration) var(--transition-easing), border var(--transition-duration) var(--transition-easing);
+}
+
+.selected {
+	background-color: var(--color-background-faded);
+}
+
+.invalid .trigger {
+	border-left-color: color(var(--color-error) saturation(-9%) lightness(+37%));
+	color: var(--color-error);
+}
+
+.valid .trigger {
+	border-left-color: color(var(--color-success) saturation(-32%) lightness(+55%));
+	color: var(--color-success);
+}
+
+.clear {
+	background: none;
+	border: var(--border-width) solid transparent;
+	color: var(--color-border);
 	cursor: pointer;
-	padding: var(--input-padding);
+	font-size: 0;
+	outline: none;
+	padding: 0;
+	position: absolute;
+	right: calc(var(--spacing-regular) * 6);
+	top: 50%;
+	transform: translateY(-50%);
 }
 
-.disabledResult {
-	color: var(--dojo-gray);
-	font-style: italic;
-	cursor: default;
+.clear:focus {
+	border-color: var(--color-highlight);
```

Comment:
I know I'm being super nitpicky (😔), but the border isn't the most beautiful of styles. Maybe Scott would have some ideas?